syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -482,11 +486,18 @@ def _public_mul(self, shares, other, equation):
                 other_is_zero = True
 
             if other_is_zero:
-                zero_shares = self.zero().child
-                return {
-                    worker: ((cmd(share, other) + zero_shares[worker]) % self.field)
-                    for worker, share in shares.items()
-                }
+                res = {}
+
+                it = iter(shares.items())
+                worker, share = next(it)
+                first_cmd = cmd(share, other)
+                zero_shares = self.zero(first_cmd.shape).child
+                res[worker] = (first_cmd + zero_shares[worker]) % self.field
+
+                for worker, share in it:
+                    cmd_res = cmd(share, other)
+                    res[worker] = (cmd(share, other) + zero_shares[worker]) % self.field
```

Comment:
Can you put all in the same loop  (and add a flag to notice when it's the first iteration) ?

Reply:
Sure